### PR TITLE
allow feed data to define attr properties

### PR DESCRIPTION
**Pull request by reubenson**
Trello: https://trello.com/c/DQdhGDDb/122-5-amazon-product-rss-feed

- Update to allow feed component to define its own rss attr fields

### CODE REVIEW

**Comment by reubenson** [comment]
@jonwinton totally agree on those points 👍

File: index.js [hunk 1]
```diff
@@ -21,7 +21,8 @@ function elevateCategory(group) {
         .filter(entry => !!entry)
```
Comment (frederickcxa) [inline index.js]: We can switch the order of these `map`/`filter`, so far we are iterating over the whole array twice as we are mapping and then filtering, if we switch them, we will only map the values that will be used.

File: index.js [hunk 2]
```diff
@@ -21,7 +21,8 @@ function elevateCategory(group) {
...
         .join(',');
     })
-    .map(string => ({ category: string }));
+    .filter(category => !!category)
```
Comment (frederickcxa) [inline index.js]: Here we can just used the `Boolean` constructor, so we have a conciser and more functional code. `.filter(Boolean)...`.

File: index.js [hunk 3]
```diff
@@ -21,7 +21,8 @@ function elevateCategory(group) {
...
+    .map(string => string ? { category: string } : {});
```
Comment (frederickcxa) [inline index.js]: It is safe to just return the object mapping, as all the empty strings are being skipped.

File: index.js [hunk 4]
```diff
@@ -62,18 +63,21 @@ function feedMetaTags({ title, description, link, copyright, generator, docs })
  * Wraps content in top level RSS and Channel tags
  *
  * @param  {Array} data
+ * @param  {Object} attr
  * @return {Object}
  */
-function wrapInTopLevel(data) {
+function wrapInTopLevel(data, attr) {
+  let defaultAttr = {
```
Comment (frederickcxa) [inline index.js]: This can be `const` as this reference does not change.

**Pull request merged**